src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -2504,25 +2528,27 @@ public bool HasBoxingConversion(TypeSymbol source, TypeSymbol destination, ref H
 
             if (source.IsRestrictedType())
             {
-                return false;
+                return ConversionKind.NoConversion;
             }
 
             if (destination.Kind == SymbolKind.DynamicType)
             {
-                return !source.IsPointerType();
+                return !source.IsPointerType()
+                    ? ConversionKind.ValueTypeBoxing
+                    : ConversionKind.NoConversion;
             }
 
             if (IsBaseClass(source, destination, ref useSiteDiagnostics))
             {
-                return true;
+                return ConversionKind.ValueTypeBoxing;
```

Comment:
this does not apply to type parameters at all.  IsBaseClass tests the actual base class, not the 'effective base class', that s type-parameter has.